src/backend/distributed/deparser/deparse_statistics_stmts.c

```diff
@@ -236,9 +236,20 @@ static void
 AppendColumnNames(StringInfo buf, CreateStatsStmt *stmt)
 {
 	ColumnRef *column = NULL;
+
 	foreach_ptr(column, stmt->exprs)
 	{
-		Assert(IsA(column, ColumnRef));
+		if (!IsA(column, ColumnRef))
+		{
+			if (stmt->if_not_exists)
+			{
+				return;
+			}
+			ereport(ERROR,
+					(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+					 errmsg(
+						 "only simple column references are allowed in CREATE STATISTICS")));
+		}
 
 		char *columnName = NameListToQuotedString(column->fields);
```

Comment:
`CreateStatistics` in postgres also checks whether the length of fields equals 1 before doing further processing